src/Crypto/Macaroon/Verifier/Internal.hs

```diff
@@ -63,15 +63,16 @@ verifySig k m = bool (Left SigMismatch) (Right m) $
     hash s c = toBytes (hmac s (vid c `BS.append` cid c) :: HMAC SHA256)
     derivedKey = toBytes (hmac "macaroons-key-generator" k :: HMAC SHA256)
 
--- | Given a list of verifiers, verify each caveat of the given macaroon
+-- | Given a list of verifiers, verify each caveat of the given
+-- list. You shouldn't use it directly unless you're doing fancy multi-step validation
 verifyCavs :: forall m pe ve. (Applicative m)
            => [Caveat -> m (VerifierResult pe ve)]
-           -> Macaroon
-           -> m (Either (ValidationError pe ve) Macaroon)
-verifyCavs verifiers m = toEither <$> errors
+           -> [Caveat]
+           -> m (Either (ValidationError pe ve) ())
```

Comment:
I'm not sure about returning `Unit` here. IMO, returning `Unit` forces you to define aliases instead of composing functions at call site.

Reply:
I don't know what we could return here, maybe `Maybe` ?

Reply:
Or maybe the list of the verified caveats ?

Reply:
Maybe we could define two functions: one that returns the caveat, and one that ignores it, like the numerous functions for `Foldable` and `Traversable`: `traverse`, `fold`, `sequenceA`, etc.

Reply:
not sure what the value is. Even exposing this function is already a stretch.

Reply:
I can return the caveats in the output, but that'd be exactly the input (if only one caveat is not verified, then we'd get a `Left`). I'm not a big fan of functions that return their arguments without modification (except for polymorphic functions of course). I can also drop this commit since it's a corner case, that nobody I know of needs. (I used to need it but found a better way to proceed)

Reply:
I see your point, and I agree with you.

We can merge this _as is_, I have no objection, you can merge it. If you want to revert that commit, I'll merge it right after my review.